src/httpation_uri_tests.erl

```diff
@@ -0,0 +1,97 @@
+-module(httpation_uri_tests).
```

Comment:
I *strongly* recommend that tests be written using a property-based testing framework and not just some hand-written handful of cases that barely scratch the surface of what might wind up getting passed through here.

For example, there is already a parsing bug in code above -- any odd-numbered case of multiple slashes (3, 5, 7, etc slashes in a row) will fail to produce the correct output, but we are required to correctly interpret and collapse this input. The tests below will not catch that, but a property-based machine generated test would -- as long as the property based test is written in a way that it knows how to generate valid, but unusual, URIs (which is not particularly difficult with PropER).